### PR TITLE
Feature/map relations metadata for metax

### PR DESCRIPTION
**Pull request by genie9**
### Description

Adds mapping for study and dataset relations: 
- study is referenced in datasets with `is_output_of` -field
- datasets are referenced in study with `relation` -field

### Related issues

Related #327 
### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (this needs a follow up PR)

### Testing

- [ ] Unit Tests
- [x] Integration Tests
- [ ] Tests do not apply
- [ ] Needs testing (start an issue or do a follow up PR about it)

### CODE REVIEW

File: metadata_backend/api/handlers/submission.py [hunk 1]
```diff
@@ -541,15 +560,19 @@ async def publish_submission(self, req: Request) -> Response:
...
+        )
+        await metax_handler.publish_dataset(metax_ids)
 
-        # Create draft DOI ??? and delete draft objects from the submission
+        # Delete draft objects from the submission
         for obj in submission["drafts"]:
             await obj_ops.delete_metadata_object(obj["schema"], obj["accessionId"])
 
-        # update study to metax with data comming from doi info
-        await metax_handler.update_dataset_with_doi_info(submission["doiInfo"], metax_ids)
```
Comment (csc-felipe) [inline metadata_backend/api/handlers/submission.py:550]: Why did you remove this line?

Comment (genie9) [reply metadata_backend/api/handlers/submission.py:550]: It was raised before the publication procedure starts to line 567, so nothing will be deleted before we know that metax is ok. Also had to extra DB call because data needed for relation mapping is added to `extra_info` -field

**Comment by genie9** [comment]
Integration tests and production deployment test has not been run for some reason. Need to investigate

**Comment by csc-felipe** [comment]
Now it's running the other tests.

**Comment by genie9** [comment]
Unfortunately not, I will update documentation and we probably need to have a chat on the subject.
